src/main/java/javaslang/collection/Stream.java

```diff
@@ -97,6 +98,20 @@
     }
 
     /**
+     * Generates an (theoretically) infinitely long Stream using a function to calculate the next value
+     * based on the previous.
+     *
+     * @param seed The first value in the Stream
+     * @param supplier A function to calculate the next value based on the previous
+     * @param <T>      value type
+     * @return A new Stream
+     */
+    static <T> Stream<T> gen(T seed, Function1<T, T> supplier) {
```

Comment:
We should use `java.util.function.Function` instead of `javaslang.Function1` here. `Function1` is a `Function`, so we widen the type and increase the interoperability with Java a little bit.